internal/directives/git_cloner.go

```diff
@@ -28,14 +29,34 @@ func init() {
 // clones one or more refs from a remote Git repository to one or more working
 // directories.
 type gitCloner struct {
+	gitUser      git.User
 	schemaLoader gojsonschema.JSONLoader
 }
 
+// gitUserFromEnv populates a git.User struct from environment variables.
+func gitUserFromEnv() git.User {
+	cfg := struct {
+		Name           string `envconfig:"GITCLIENT_NAME"`
+		Email          string `envconfig:"GITCLIENT_EMAIL"`
+		SigningKeyType string `envconfig:"GITCLIENT_SIGNING_KEY_TYPE"`
+		SigningKeyPath string `envconfig:"GITCLIENT_SIGNING_KEY_PATH"`
+	}{}
+	envconfig.MustProcess("", &cfg)
+	return git.User{
+		Name:           cfg.Name,
+		Email:          cfg.Email,
+		SigningKeyType: git.SigningKeyType(cfg.SigningKeyType),
+		SigningKeyPath: cfg.SigningKeyPath,
+	}
+}
```

Comment:
The retired git-based promo mechs were configured similarly and this was accidentally not included in #2537